### PR TITLE
C# 8.0: stackalloc in nested expressions

### LINKED ISSUE #14489: C# 8.0: stackalloc in nested contexts
dotnet/csharplang#1412

@BillWagner this is for the [C# 8 docs project](https://github.com/dotnet/docs/projects/58).
Could you please assign this issue to me?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ecde487e-96e5-85c2-ff87-227994f705ad
* Version Independent ID: f77588b3-9b31-f8ce-fad5-bf937337eb8c
* Content: [stackalloc operator - C# reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc)
* Content Source: [docs/csharp/language-reference/operators/stackalloc.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/operators/stackalloc.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by pkulikov**
Fixes #14489
Supported by dotnet/samples#1526
/cc: @gafter

### CODE REVIEW

File: docs/csharp/whats-new/csharp-8.md [hunk 1]
```diff
@@ -488,6 +489,16 @@ Span<Coords<int>> coordinates = stackalloc[]
 
 For more information, see [Unmanaged types](../language-reference/builtin-types/unmanaged-types.md).
 
+## stackalloc in nested expressions
+
+Starting with C# 8.0, if the result of a [stackalloc](../language-reference/operators/stackalloc.md) expression is of the <xref:System.Span%601?displayProperty=nameWithType> or <xref:System.ReadOnlySpan%601?displayProperty=nameWithType> type, you can use the `stackalloc` expression in other expressions:
+
+```csharp
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-8.md]: I think this should be referenced from the samples repo ?

Comment (pkulikov) [reply docs/csharp/whats-new/csharp-8.md]: To be consistent with all other snippets in this article, this one is embedded in the file. Furthermore, as I understand any snippet should be referenced only by one article at most (then one can safely edit the snippet according to the article context). So, referencing this one would require creating a new project for the snippets in this article in the samples repo. While that may be a good move (so, I agree with you @Youssef1313) , I think it's not in the scope of this PR.

Comment (Youssef1313) [reply docs/csharp/whats-new/csharp-8.md]: > any snippet should be referenced only by one article at most (then one can safely edit the snippet according to the article context).

Good point, I didn't know that.

Comment (BillWagner) [reply docs/csharp/whats-new/csharp-8.md]: > any snippet should be referenced only by one article at most (then one can safely edit the snippet according to the article context).

This is something we'd like to change in the future, with better navigation between samples and articles. /cc @dend

**Review by BillWagner — APPROVED** [review]
Thanks for adding this @pkulikov 

I'll :shipit: now.

**Pull request merged**